### PR TITLE
Hw 3 book store

### PR DESCRIPTION
**Pull request by dmytrovb02**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -46,20 +70,46 @@
 			<scope>test</scope>
 		</dependency>
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-validation</artifactId>
+			<groupId>org.hibernate.validator</groupId>
+			<artifactId>hibernate-validator</artifactId>
```
Comment (sokimaaa) [inline pom.xml]: use starter instead

File: src/main/java/com/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -1,7 +1,52 @@
...
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookResponseDto getBookById(@PathVariable Long id) {
+        return bookService.findById(id);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
```
Comment (sokimaaa) [inline src/main/java/com/bookstore/controller/BookController.java]: Use ResponseEntity<?> instead
```suggestion
```

File: src/main/java/com/bookstore/service/BookService.java [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
 
 public interface BookService {
 
-    Book save(Book book);
+    BookResponseDto create(BookRequestDto bookRequestDto);
 
-    List<Book> findAll();
+    List<BookResponseDto> findAll();
+
+    BookResponseDto findById(Long id);
```
Comment (sokimaaa) [inline src/main/java/com/bookstore/service/BookService.java:13]: Use Optional
```suggestion
    BookResponseDto findById(Long id);
```

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -33,10 +34,27 @@ public List<BookResponseDto> findAll() {
     }
 
     @Override
-    public BookResponseDto findById(Long id) {
+    public Optional<BookResponseDto> findById(Long id) {
         return bookRepository.findById(id)
```
Comment (Rommelua) [inline src/main/java/com/bookstore/service/impl/BookServiceImpl.java]: throw EntityNotFoundException if bookRepository returns empty Optional

Comment (dmytrovb02) [reply src/main/java/com/bookstore/service/impl/BookServiceImpl.java]: In this way?

**Review by Rommelua — APPROVED** [review]

File: src/main/java/com/bookstore/service/impl/BookServiceImpl.java (hunk 4 above)
Comment (Rommelua) [reply src/main/java/com/bookstore/service/impl/BookServiceImpl.java]: yes

**Pull request merged**